.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,96 @@
+name: Build and Test
+
+concurrency:
+  group: ${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+  push:
+    branches:
+      - master
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-22.04
+    timeout-minutes: 10
+    steps:
+
+    - name:
+      uses: actions/checkout@v3
+
+    - name: Build
+      run: go build -v ./...
+
+    - name: Test
+      run: go test -v -race -bench=. -benchtime=100ms ./...
+
+  linux:
+    name: Linux Build
+    runs-on: ubuntu-22.04
+    timeout-minutes: 10
+    needs: test
+    steps:
+      - name:
+        uses: actions/checkout@v3
+
+      - name: Build
+        run: make linux
+
+  windows:
+    name: Windows Build
+    runs-on: ubuntu-22.04
+    timeout-minutes: 30
+    needs: test
+    steps:
+      - name:
+        uses: actions/checkout@v3
+
+      - name: Build
+        run: make windows
```

Comment:
This pulls a gigantic Docker image. I wonder if we should make the windows binary in the same container as linux to save a pull.

Reply:
Alternatively, we could just call `go build` from a linux or windows machine, which would be a lot cheaper and faster. The huge image is to get the cross-compile tools, which we don't need if we are already in the platform.

Reply:
> I wonder if we should make the windows binary in the same container as linux to save a pull.

OK, done.  The Linux and Windows builds take about 2 minutes each separately, and 

> we could just call `go build`

This was already in place as part of the test automation.  This PR is about testing the Makefile, basically.  (Also, my next step is to add release automation, so we don't have to copy four different artifacts to outline-client and risk a repeat of the v1.9.0 Windows issue.)